packages/virtual-core/src/index.ts

```diff
@@ -464,16 +467,64 @@ export class Virtualizer<
 
       for (let i = min; i < count; i++) {
         const key = getItemKey(i)
-        const measuredSize = itemSizeCache.get(key)
-        const start = measurements[i - 1]
-          ? measurements[i - 1]!.end
+
```

Comment:
I would do this calculations in condition

```tsx
const measuredSize = itemSizeCache.get(key)
const size =
  typeof measuredSize === 'number'
    ? measuredSize
    : this.options.estimateSize(i)

if (this.options.columns > 1) {
  // ... calculations

  const start = furthestMeasurement
    ? furthestMeasurement.end
    : paddingStart + scrollMargin

  const end = start + size

  const column = furthestMeasurement
    ? furthestMeasurement.column
    : i % this.options.columns

  measurements[i] = {
    index: i,
    start,
    size,
    end,
    key,
    column,
  }
} else {
  const start = measurements[i - 1]
    ? measurements[i - 1]!.end
    : paddingStart + scrollMargin
  const end = start + size

  measurements[i] = {
    index: i,
    start,
    size,
    end,
    key,
    column: 1,
  }
}
```

Reply:
Or even better lift the check `this.options.lanes > 1` before the for loop, i think here is pretty acceptable to have this small duplication. Overall @zhouzi awesome work 👏

Reply:
The only difference between the two code branches is how the previous measurement is found. So I added a method that follows your suggestion by returning `measurements[i - 1]` when lanes === 1. No code duplication 👍

Reply:
Ok, that work also, thanks 👍 Imho having check for `lanes` before calling the method would better, as there is no need to call it, also logic is one place.

Will test it a bit and lets merge it 💪

Reply:
> Imho having check for `lanes` before calling the method would better, as there is no need to call it, also logic is one place.

Done, I moved the condition out of the method 😉